### PR TITLE
libtomcrypt: added v1.18.2

### PR DESCRIPTION
**Pull request by jowr**
**libtomcrypt/1.18.2**

This is an established cryptographic library and I have several projects that use it. I am sure that others also find this package useful given the popularity this library. See https://github.com/libtom/libtomcrypt for details.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/libtomcrypt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            os.rename(os.path.join(self.package_folder, "lib", "tomcrypt.dll.lib"),
+                      os.path.join(self.package_folder, "lib", "tomcrypt.lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["tomcrypt"]
+        if not self.options.shared:
```
Comment (prince-chrismc) [inline recipes/libtomcrypt/all/conanfile.py:141]:
```suggestion
        self.cpp_info.libs = ["tomcrypt"]
        self.cpp_info.set_property("pkg_config_name", "libtomcrypt")
        if not self.options.shared:
```

File: recipes/libtomcrypt/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            if self.settings.os == "Windows":
+                self.cpp_info.system_libs = ["advapi32", "crypt32"]
+
+        self.cpp_info.names["pkg_config"] = "libtomcrypt"
```
Comment (prince-chrismc) [inline recipes/libtomcrypt/all/conanfile.py:145]:
```suggestion
        # TODO: to remove in conan v2 once pkg_config generators removed
        self.cpp_info.names["pkg_config"] = "libtomcrypt"
```

**Review by prince-chrismc — COMMENTED** [review]
It's really good! Just one nit picky thing to future proof

**Comment by jowr** [comment]
Thanks for reviewing!

**Comment by prince-chrismc** [comment]
Why does windows fail? 😖 

https://github.com/conan-io/conan-center-index/pull/9066/files/e1070dab2d8d247aa8fd1689dc818ee383d57591..c76a50df9d9863e19d358fbf61b4931d4e3265c2 should have changes anything...

Looking at it seems like a one off random

https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/9066/6-configs/windows-visual_studio/libtomcrypt/1.18.2//summary.json

Maybe retrigger CI, close the pr wait 10s and then re-open it 🔁 if not I'll ask for help

**Comment by shiena** [comment]
I have found similar error issue. https://github.com/conan-io/conan-center-index/issues/9285
Does `cmake_find_package` seem relevant?

**Comment by prince-chrismc** [comment]
Sadly no, the test package does not use CMake, it seems there's more work required to get the windows shared exports to work correctly 😢 the `lib` is missing symbols

**Comment by shiena** [comment]
@jowr 
Diff to build test_package with cmake instead of pkg-config. Does this seem to build?

```diff
diff --git a/recipes/libtomcrypt/all/test_package/CMakeLists.txt b/recipes/libtomcrypt/all/test_package/CMakeLists.txt
index c13db6494..6c6db0de2 100644
--- a/recipes/libtomcrypt/all/test_package/CMakeLists.txt
+++ b/recipes/libtomcrypt/all/test_package/CMakeLists.txt
@@ -2,11 +2,10 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
 include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
-conan_basic_setup(TARGETS)
+conan_basic_setup()
 
-include(FindPkgConfig)
-pkg_check_modules(LibTomCrypt REQUIRED IMPORTED_TARGET libtomcrypt)
+find_package(LibTomCrypt REQUIRED)
 
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LibTomCrypt)
+target_link_libraries(${PROJECT_NAME} PRIVATE ${CONAN_LIBS})
 target_compile_definitions(${PROJECT_NAME} PRIVATE LTC_NO_PROTOTYPES)
diff --git a/recipes/libtomcrypt/all/test_package/conanfile.py b/recipes/libtomcrypt/all/test_package/conanfile.py
index 9e09e219f..3da371b66 100644
--- a/recipes/libtomcrypt/all/test_package/conanfile.py
+++ b/recipes/libtomcrypt/all/test_package/conanfile.py
@@ -4,10 +4,7 @@ import os
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "pkg_config"
-
-    def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+    generators = "cmake", "cmake_find_package"
 
     def build(self):
         cmake = CMake(self)
```

**Pull request closed without merging**